docs/nginx-setup.md

```diff
@@ -0,0 +1,74 @@
+# Setting up LetsEncrypt Certificate with Nginx Reverse Proxy
+
+- setup virtual host config as in `./nginx.conf`
+
+- create a softlink
+```
+  sudo ln -s /etc/nginx/sites-available/subsites /etc/nginx/sites-enabled/subsites
+```
+
+- get a cert using LetsEncrypt (Make sure LetsEncrypt is installed)
+```
+sudo letsencrypt certonly -a webroot --webroot-path=/var/www/html -d hub.gdgx.io
+
+# if letencrypt was installed via git
+./letsencrypt-auto certonly -a standalone -d hub.gdgx.io
+```
+
+- check if certs are present
+```
+sudo ls -l /etc/letsencrypt/live/hub.gdgx.io
+```
+
+- create a secure private key
+```
+sudo openssl dhparam -out /etc/ssl/certs/dhparam.pem 2048
+```
+
+- create SSL cert conf file for nginx
+```
+sudo nano /etc/nginx/snippets/ssl-hub.gdgx.io.conf
+
+## in the file
+ssl_certificate /etc/letsencrypt/live/hub.gdgx.io/fullchain.pem;
+ssl_certificate_key /etc/letsencrypt/live/hub.gdgx.io/privkey.pem;
+```
+
+- create SSL params conf for nginx
+```
+sudo nano /etc/nginx/snippets/ssl-params.conf
+
+## in the file
+# from https://cipherli.st/
+# and https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html
```

Comment:
Can we split this into two blocks? One for the command one for the file contents.